client/js/ingester.coffee

```diff
@@ -5,18 +5,54 @@
   chromeIngest: (event) ->
     items = event.originalEvent.dataTransfer.items
 
-    j = 0
-    while j < items.length
-      item = items[j].webkitGetAsEntry()
+    for item in items
+      item = item.webkitGetAsEntry()
       if item
-        console.log(item)
-      j++
+        @_chromeAddItem(item)
 
   ingest: (event) ->
-    # Other browser users have to upload files directly
     files = event.originalEvent.dataTransfer.files
 
-    j = 0
-    while j < files.length
-      if files[j].type.match(/audio\/(mp3|mpeg)/)
-        console.log(files[j])
+    for file in files
+      @_addFile(file)
+
+  _chromeAddItem: (item, path) ->
+    path = path || ""
+
+    if item.isFile
+      item.file (file) =>
+        @_addFile(file)
+
+    else if item.isDirectory
+      dirReader = item.createReader()
+      dirReader.readEntries (entries) ->
+        for entry in entries
+          @_chromeAddItem(entry, path + item.name + "/")
+
+  _addFile: (file) ->
+    jsmediatags.read(
+      file,
+      {
+        onSuccess: (data) =>
+          @_uploadFile(file).then( ->
+            Meteor.call('addSong', data, file.name)
+          )
+        onError: ->
+          alert("Not a valid file")
+      }
+    )
+
+  _uploadFile: (file) ->
+    return new Promise (resolve, reject) ->
+      bucket = new AWS.S3
+      params = {
+        ACL: "public-read",
+        Body: file,
+        Bucket: Meteor.settings.public.bucketName
+        ContentType: file.type,
+        Key: file.name,
```

Comment:
should the key identify the user? or will different users share the same actual file in s3?